File: Java/Shopping.java
import java.util.*;
import java.io.*;

public class Shopping {

	static int maxToys(List<Integer> cost, int k){

		int sum = 0;
		int count = 0;

		for (int c : cost) {
			if (sum + c <= k){
				sum += c;
				count++;
			}

		}
		cost.forEach( name ->
			System.out.println("$" + name));


		return count;
	}

	public static void main(String args[]){
		List<Integer> cost = Arrays.asList(10, 12, 15, 20, 25, 30, 40, 50, 21, 17);
		cost.sort(null);
		System.out.print(cost + "\n");
		int k = 100;

		System.out.print("You can buy " + maxToys(cost, k) + " Toys with $" + k +"\n");
	}

}

File: Go/shopping.go
package main


import (
	"fmt"
)


func main(){

	var k int

	cost := [] int{10,20,30,40,50,70,90}

	fmt.Println("Please enter the amount you have to spend: ")
	_, err := fmt.Scanf("%d", &k)

	if err != nil {
		fmt.Println(err)
	}

	fmt.Println(maxToys(cost, k))
}


func maxToys(cost []int, K int) int {
	
	sum := 0
	count := 0

	for _, v := range(cost){
		if sum + v <= K {
			sum  = sum + v
			count++
		}
	}
	return count
}
